lib.rs

```diff
@@ -245,6 +310,9 @@ impl<A: Array> SmallVec<A> {
         }
     }
 
+    /// Reserve capacity for at least `additional` more elements to be inserted.
+    ///
+    /// May reserve more space to avoid frequent reallocations.
```

Comment:
Maybe make a note about overflow behaviour. In particular, it is not safe to assume that you can actually fit `additional` new elements in it!